plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/AbstractContractStatechartGeneration.xtend

```diff
@@ -466,13 +467,48 @@ abstract class AbstractContractStatechartGeneration {
 		}
 	}
 
-	def protected setupForwardTransition(ModalInteractionSet set, boolean reversed, boolean isNegated,
-		Transition forwardTransition) {
-		val eventParamRefs = ecoreUtil.getAllContentsOfType(set, EventParameterReferenceExpression)
+	def protected retargetAllEventParamRefs(EObject container, boolean reversed) {
+		val eventParamRefs = ecoreUtil.getAllContentsOfType(container, EventParameterReferenceExpression)
 		for (eventParamRef : eventParamRefs) {
 			eventParamRef.port = getPort(getNameOfNewPort(eventParamRef.port, reversed))
 			eventParamRef.event = getEvent(eventParamRef.event.name, eventParamRef.port)
 		}
+	}
+
+	def protected addAssignmentsToTransition(Iterable<ScenarioAssignmentStatement> assignments, Transition transition) {
+		for (assignment : assignments) {
```

Comment:
Why not a single line: `transition.effects += assignments.clone`?

Reply:
This is how I tired it first, but I received a parsing exception